account_invoice_block_payment/models/account_move.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    payment_blocked = fields.Boolean(
+        help="if true, the payment of the invoice is blocked.",
+    )
+
+    def action_register_payment(self):
+        error_msg = ""
+        for move in self:
+            if move.payment_blocked:
+                error_msg += (
+                    _("The payment on invoice {} is blocked.").format(move.name) + "\n"
+                )
+        if len(error_msg):
```

Comment:
```suggestion
        if error_msg:
```

In Python a character string which is not empty is considered as True